src/idpLoader.ts

```diff
@@ -0,0 +1,102 @@
+import nodeFetch from "node-fetch";
+import { DOMParser } from "xmldom";
+import { log } from "./utils/logger";
+
+export interface IDPMetadataOptions {
+  cert: ReadonlyArray<string>;
+  entryPoint: string;
+  logoutUrl: string;
+}
+
+interface IDPMetadataParams extends IDPMetadataOptions {
+  entityID: string;
+}
+
+export class IDPMetadata {
+  public cert: ReadonlyArray<string>;
+  public entityID: string;
+  public entryPoint: string;
+  public logoutUrl: string;
+  constructor(params: IDPMetadataParams) {
+    this.cert = params.cert;
+    this.entityID = params.entityID;
+    this.entryPoint = params.entryPoint;
+    this.logoutUrl = params.logoutUrl;
+  }
+
+  public getIDPOption(): IDPMetadataOptions {
+    return {
+      cert: Array.from(this.cert),
+      entryPoint: this.entityID,
+      logoutUrl: this.logoutUrl
+    };
+  }
+}
+
+export const ARUBA_ID = "https://loginspid.aruba.it";
```

Comment:
why declaring these IDs here if this file doesn't use them but only one other file imports them? I suggest to move them to `spidStrategy.ts`

Reply:
fixed in commit ebb3cf5